media-libs/boca/boca-1.0~alpha20180913.recipe

```diff
@@ -0,0 +1,69 @@
+SUMMARY="An audio component framework and library"
+DESCRIPTION="BoCA is the component framework behind the fre:ac audio \
+converter. It provides unified interfaces for components like encoders, \
+decoders, taggers and extensions as well as code to support communication \
+between applications and their components."
+HOMEPAGE="https://github.com/enzo1982/BoCA"
+COPYRIGHT="2007-2018 Robert Kausch"
+LICENSE="GNU GPL v2"
+REVISION="1"
+SOURCE_URI="https://github.com/enzo1982/BoCA/releases/download/v1.1-alpha-20180913/freac-cdk-1.1-alpha-20180913.tar.gz"
+CHECKSUM_SHA256="d8a6f59e3ccc68b3dbe5d77e5f3e97240691196467656714c678ccbf2932257b"
+SOURCE_DIR="freac-cdk-1.1-alpha-20180913"
+
+ARCHITECTURES="!x86_gcc2 x86 x86_64"
+
+libVersion="${portVersion%~*}"
+libVersionCompat="0 compat >= 0"
+
+PROVIDES="
+	boca$secondaryArchSuffix = $portVersion
+	lib:libboca_${libVersion}$secondaryArchSuffix = $libVersionCompat
+	"
+REQUIRES="
+	haiku$secondaryArchSuffix
+	lib:libsmooth_0.8.74$secondaryArchSuffix >= 0
+	lib:libexpat$secondaryArchSuffix
+	lib:libz$secondaryArchSuffix
+	"
+
+PROVIDES_devel="
+	boca${secondaryArchSuffix}_devel = $portVersion
+	devel:libboca_${libVersion}$secondaryArchSuffix = $libVersionCompat
+	"
+REQUIRES_devel="
+	boca$secondaryArchSuffix == $portVersion base
+	devel:libsmooth_0.8.74$secondaryArchSuffix >= 0
+	"
+
+BUILD_REQUIRES="
+	haiku${secondaryArchSuffix}_devel
+	devel:libsmooth_0.8.74$secondaryArchSuffix
+	devel:libexpat$secondaryArchSuffix
+	devel:libz$secondaryArchSuffix
+	"
+BUILD_PREREQUIRES="
+	cmd:g++$secondaryArchSuffix
+	cmd:gcc$secondaryArchSuffix
+	cmd:ld$secondaryArchSuffix
+	cmd:make
+	cmd:pkg_config$secondaryArchSuffix
+	"
+
+BUILD()
+{
+	make $jobArgs
+}
+
+INSTALL()
+{
+	make prefix="$prefix" libdir="$libDir" includedir="$includeDir" install
+
+	mkdir -p "$developLibDir"
+
+	ln -s "$libDir"/libboca-${libVersion}.so.0 "$developLibDir"/libboca-${libVersion}.so
+	ln -s "$libDir"/libboca-${libVersion}.so.0 "$developLibDir"/libboca-${libVersion}.so.0
```

Comment:
Hmm, did you try with prepareInstalledDevel**Lib** or prepareInstalledDevel**Libs**?

* prepareInstalledDevel**Lib** works for one lib, but accepts an optional second argument which is the alternate name that will be given to the symlink, if I'm not mistaken.
* prepareInstalledDevel**Libs** works for any number of libs but, obviously, does not accept any optional argument. Of course, it can also be used with one lib.

I'm asking this question because if you have 2 libs but call prepareInstalledDevel**Lib** (nistead of prepareInstalledDevel**Libs**) then you don't get what you expect.

BTW, there is currently no policy about which should be used when there is only one lib. I personally prefer to use prepareInstalledDevel**Lib** in such cases (because prepareInstalledDevel**Libs** just calls prepareInstalledDevel**Lib** for each of its arguments), but several contributors prefer to always use prepareInstalledDevel**Libs**.

Anyway, here I guess you would have to use prepareInstalledDevel**Libs** if you want to pass two libraries. Well, I don't know how this macro works if you call it with the basename of a lib and the basename of a symlink to the lib.

Reply:
Changed to relative links, thanks! 😃 Also removed the curly braces.

Reply:
Didn't have luck with either prepareInstalledDevel**Lib** or prepareInstalledDevel**Libs**.

The scripts seem to try to find a shared library, but can find only symlinks in BoCA's case (as the actual library is in `lib/boca`).

So I'd like to keep creating the links manually for now.

Reply:
I've checked this before and (don't do :) ) prepareInstalledDevelLib boca/boca worked, but it moved the entire folder to the devel package (leaving no lib in the base package)